packages/teuchos/comm/src/Teuchos_TimeMonitor.hpp

```diff
@@ -192,8 +192,11 @@ class TEUCHOSCOMM_LIB_DLL_EXPORT TimeMonitor :
   ///   Default behavior is not to reset the timer.
   TimeMonitor (Time& timer, bool reset=false);
 
+  //! Default constructor is deleted, since it would be unsafe.
+  TimeMonitor () = delete;
+
   //! Destructor: stops the timer.
-  ~TimeMonitor();
+  ~TimeMonitor() override;
```

Comment:
Should this not also have 'virtual' to make this more clear?

Reply:
[C++ Core Guidelines C.128](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rh-override) says: "Virtual functions should specify exactly one of `virtual`, `override`, or `final`."

Rationale is that specifying both `virtual` and `override` is redundant.  See the example [https://en.cppreference.com/w/cpp/language/virtual](here): "Then this function in the class `Derived` is also virtual (whether or not the keyword `virtual` is used in its declaration) and overrides `Base::vf` (whether or not the word `override` is used in its declaration)."

I realize now that C.128 says that `TimeMonitor`'s destructor should be declared neither `virtual` nor `override`.  It's harmless to leave `override` in place, though.  We do that a lot, and I suspect some tools or compiler flags will warn if child class destructors lack decoration.